cmd/cosign/cli/sign/sign.go

```diff
@@ -151,6 +151,10 @@ func SignCmd(ro *options.RootOptions, ko options.KeyOpts, regOpts options.Regist
 			return fmt.Errorf("unable to resolve attachment %s for image %s", attachment, inputImg)
 		}
 
+                if _, ok := ref.(name.Tag); ok {
+                   fmt.Println("Warning: Tag used in reference to identify the image. Consider supplying the digest for immutability.")
```

Comment:
If we print the warning after we resolve the digest, the error message can give you exactly the command to run instead, without needing a placeholder digest.

Reply:
> If we print the warning after we resolve the digest, the error message can give you exactly the command to run instead, without needing a placeholder digest.

I was thinking this... but by telling the user the digest they should plan to use... doesn't that hold them susceptible for the very problem that the warning tries to avoid? Instead the warning should say "make sure you figure out what the digest of the image is, but we are not the ones to tell you now, just know for next time". Maybe I have other thought it @imjasonh

Reply:
+1 to @ChaosInTheCRD here.

BTW I'm following up in https://github.com/sigstore/cosign/pull/2313 so we can have discussion there.